lib/shopify_api/api_version.rb

```diff
@@ -1,113 +1,111 @@
 # frozen_string_literal: true
 module ShopifyAPI
-  class ApiVersion
+  class ApiVersion < ActiveResource::Base
     class UnknownVersion < StandardError; end
     class InvalidVersion < StandardError; end
-
-    extend DefinedVersions
-
     include Comparable
 
-    def self.coerce_to_version(version_or_name)
-      return version_or_name if version_or_name.is_a?(ApiVersion)
-
-      @versions ||= {}
-      @versions.fetch(version_or_name.to_s) do
-        raise UnknownVersion, "#{version_or_name} is not in the defined version set: #{@versions.keys.join(', ')}"
+    self.site = "https://app.shopify.com/services/"
+    self.primary_key = :handle
+
+    HANDLE_FORMAT = /^\d{4}-\d{2}$/.freeze
+    UNSTABLE_HANDLE = 'unstable'
+    UNSTABLE_AS_DATE = Time.utc(3000, 1, 1)
+    API_PREFIX = '/admin/api/'
+
+    class << self
+      attr_reader :versions
+
+      def coerce_to_version(version_or_handle)
+        return version_or_handle if version_or_handle.is_a?(ApiVersion)
+        if @versions.nil?
+          warn "[API VERSION WARNING] Known API Version set is empty. Initializing unvalidated version from handle."
```

Comment:
I think we can come up with a warning message that will fit better in the context of what an app developer cares about?

Reply:
I think so. I was drawing a blank though. Any suggestions?

Reply:
"Unable to validate API Version"?